addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/discovery/LGWebOSDiscovery.java

```diff
@@ -0,0 +1,223 @@
+        Util.uninit();
+    }
+
+    // @Override
+    @Override
+    protected void startScan() {
+        // no adhoc scanning. Discovery Service runs in background, but re-discover all known devices incase they were
+        // deleted from the inbox.
+        for (ConnectableDevice device : discoveryManager.getCompatibleDevices().values()) {
+            thingDiscovered(createDiscoveryResult(device));
+        }
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        discoveryManager.start();
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        discoveryManager.stop();
+    }
+
+    // DiscoveryManagerListener
+
+    @Override
+    public void onDeviceAdded(DiscoveryManager manager, ConnectableDevice device) {
+        thingDiscovered(createDiscoveryResult(device));
+    }
+
+    @Override
+    public void onDeviceUpdated(DiscoveryManager manager, ConnectableDevice device) {
+        logger.debug("Device updated: {}", device);
+        thingRemoved(createThingUID(device));
+        thingDiscovered(createDiscoveryResult(device));
+    }
+
+    @Override
+    public void onDeviceRemoved(DiscoveryManager manager, ConnectableDevice device) {
+        logger.debug("Device removed: {}", device);
+        thingRemoved(createThingUID(device));
+    }
+
+    @Override
+    public void onDiscoveryFailed(DiscoveryManager manager, ServiceCommandError error) {
+        logger.warn("Discovery Failed {}", error.getMessage());
+    }
+
+    // Helpers for DiscoveryManagerListener Impl
+    private DiscoveryResult createDiscoveryResult(ConnectableDevice device) {
+        ThingUID thingUID = createThingUID(device);
+        return DiscoveryResultBuilder.create(thingUID).withLabel(device.getFriendlyName()).build();
+    }
+
+    private ThingUID createThingUID(ConnectableDevice device) {
+        return new ThingUID(THING_TYPE_WEBOSTV, device.getId());
+    }
+
+    public DiscoveryManager getDiscoveryManager() {
+        return this.discoveryManager;
+    }
+
+    @Override
+    public String getDataDir() {
+        return ConfigConstants.getUserDataFolder() + File.separator + "lgwebos";
+    }
+
+    @Override
+    public String getApplicationName() {
+        return "openHAB";
+    }
+
+    @Override
+    public String getPackageName() {
+        // In combination with ApplicationName this is used by LG TVs to identify the client application.
+        // We don't want the actual package name. We need a constant namespace here.
+        return "org.openhab";
+    }
+
+    @Override
+    public InetAddress getIpAddress() {
+        return localInetAddresses;
+    }
+
+    /**
+     * Get local IP either through configuration or auto detection.
+     * Method will ignore loopback addresses.
+     *
+     * @param localIP optional configuration string
+     * @return local ip or <code>null</code> if detection was not possible.
+     */
+    private InetAddress findLocalInetAddresses(String localIP) {
```

Comment:
You should not implement this yourself, but rather use the`org.eclipse.smarthome.core.net.NetworkAddressService` instead. This system service is already configurable and the user can set his local IP, which can then be used by ALL bindings, so there is no need for the binding to have such a configuration parameter at all.

Reply:
Perfect.

Reply:
I realize this is not yet available in my 2.1 installation.
But it works on 2.2.